addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -533,22 +546,133 @@ var ImageWidget = MediaWidget.extend({
             }
 
             function _processFile(attachment, error) {
-                var $button = self.$('.o_upload_image_button');
                 if (!error) {
-                    $button.addClass('btn-success');
                     self._toggleImage(attachment, true);
                 } else {
-                    $button.addClass('btn-danger');
                     self.$el.addClass('o_has_error').find('.form-control, .custom-select').addClass('is-invalid');
                     self.$el.find('.form-text').text(error);
                 }
-
-                if (!self.multiImages) {
-                    self.trigger_up('save_request');
-                }
             }
         });
         this.$el.submit();
+        // Emptying file inputs
+        this.$('#imagefile').val('');
+        this.$('#documentfile').val('');
```

Comment:
Is this useful ? (real question, I do not know)

Reply:
Indeed, this seems not useful. I was trying things when was looking at document upload issue.. :smiley:

Reply:
@qsm-odoo In fact it is useful. If you upload an image, and then if you add an image via URL this time, you'll get the previous image (not the one from the URL). Emptying file inputs is fixing this.

Reply:
Ok, but then, for a particular `ImageWidget` there is only one file input not two, and we don't have to care if we are using a document or an image at that moment. So something like
```javascript
this.$('.o_file_input').val('');
```